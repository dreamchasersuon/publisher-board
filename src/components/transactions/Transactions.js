import React from 'react';
import './Transactions.css';
import Transaction from './transaction/Transaction';
import { useSelector } from 'react-redux';

export default function Transactions() {
  const { transactions } = useSelector(state => state.settings);

  return (
    <div data-testid="modal-transactions" className="Transactions">
      {transactions
        ? transactions.map(transaction => {
            return (
              <Transaction
                key={transaction.operation_id}
                transaction={transaction}
              />
            );
          })
        : 'Transactions for this period is not found'}
    </div>
  );
}
