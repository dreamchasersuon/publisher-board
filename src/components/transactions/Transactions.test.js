import React from 'react';
import { Provider } from 'react-redux';
import snapshotDiff from 'snapshot-diff';
import Transactions from './Transactions';
import configureStore from 'redux-mock-store';

const transactions = [
  {
    amount: 12,
    comment: null,
    date: '2018-12-28T15:02:29+03:00',
    operation_id: 178199125,
    virtual_items: [
      {
        amount: 1,
        virtual_item: {
          localized_name: 'Item2',
          sku: '002',
        },
      },
      {
        amount: 123,
        virtual_item: {
          localized_name: 'Item3',
          sku: '003',
        },
      },
    ],
  },
  {
    amount: 1200,
    comment: 'Update balance',
    date: '2017-02-10T15:02:29+03:00',
    operation_id: 828328435,
  },
];

test('renders transactions or not found if empty', () => {
  const mockStore = configureStore([]);

  expect(
    snapshotDiff(
      <Provider store={mockStore({ settings: { transactions } })}>
        <Transactions />
      </Provider>,
      <Provider store={mockStore({ settings: { transactions: [] } })}>
        <Transactions />
      </Provider>,
    ),
  ).toMatchSnapshot();
});
