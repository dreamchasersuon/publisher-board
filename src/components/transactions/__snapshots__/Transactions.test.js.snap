// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders transactions or not found if empty 1`] = `
"Snapshot Diff:
- <Provider store={{\\"clearActions\\": [Function clearActions], \\"dispatch\\": [Function dispatch], \\"getActions\\": [Function getActions], \\"getState\\": [Function getState], \\"replaceReducer\\": [Function replaceReducer], \\"subscribe\\": [Function subscribe]}}><Transactions /></Provider>
+ <Provider store={{\\"clearActions\\": [Function clearActions], \\"dispatch\\": [Function dispatch], \\"getActions\\": [Function getActions], \\"getState\\": [Function getState], \\"replaceReducer\\": [Function replaceReducer], \\"subscribe\\": [Function subscribe]}}><Transactions /></Provider>

  <div
    className=\\"Transactions\\"
    data-testid=\\"modal-transactions\\"
- >
-   <div
-     className=\\"Transaction\\"
-   >
-     <span
-       className=\\"Transaction-date\\"
-     >
-       28 Nov 2018
-     </span>
-     <div
-       className=\\"Transaction-item\\"
-     >
-       <span
-         className=\\"TransactionItem-name\\"
-       >
-         Item2
-       </span>
-       <span
-         className=\\"TransactionItem-amount\\"
-       >
-         1
-       </span>
-     </div>
-     <div
-       className=\\"Transaction-item\\"
-     >
-       <span
-         className=\\"TransactionItem-name\\"
-       >
-         Item3
-       </span>
-       <span
-         className=\\"TransactionItem-amount\\"
-       >
-         123
-       </span>
-     </div>
-     <div
-       className=\\"Transaction-comment_and_amount\\"
-     >
-       <span
-         className=\\"Transaction-comment\\"
  />
-       <span
-         className=\\"Transaction-total\\"
-       >
-         Total: 
-         124
-       </span>
-     </div>
-   </div>
-   <div
-     className=\\"Transaction\\"
-   >
-     <span
-       className=\\"Transaction-date\\"
-     >
-       10 Jan 2017
-     </span>
-     <div
-       className=\\"Transaction-comment_and_amount\\"
-     >
-       <span
-         className=\\"Transaction-comment\\"
-       >
-         Update balance
-       </span>
-       <span
-         className=\\"Transaction-total\\"
-       >
-         Total: 
-         1200
-       </span>
-     </div>
-   </div>
- </div>"
`;
